python/lsst/ts/fiberspectrograph/dataManager.py

```diff
@@ -107,18 +108,26 @@ def make_fits_header(self, data):
         # as comments on each of these, but pydoc can't see them.
         hdr["FORMAT_V"] = FORMAT_VERSION
         hdr["OBSERVAT"] = "Vera C. Rubin Observatory"
-        hdr["INSTRUME"] = self.instrument
-        hdr["ORIGIN"] = self.origin
-        hdr["LOCATN"] = (None, "Location of Instrument")
+        hdr["INSTRUME"] = (self.instrument, "Type of Instrument.")
+        hdr["SERIAL"] = (self.serial, "Serial Number of Spectrograph.")
+        hdr["ORIGIN"] = (self.origin, "Name of the program that produced this data.")
+        hdr["LOCATN"] = (None, "Location of Instrument.")
         hdr["DETSIZE"] = data.n_pixels
         hdr["DATE-BEG"] = astropy.time.Time(data.date_begin).tai.fits
         hdr["DATE-END"] = astropy.time.Time(data.date_end).tai.fits
-        hdr["EXPTIME"] = data.duration
+        hdr["DAYOBS"] = astropy.time.Time(data.date_begin).tai.strftime("%Y%m%d")
+        hdr["EXPTIME"] = (data.duration, "Duration of scan.")
         hdr["TIMESYS"] = "TAI"
-        hdr["IMGTYPE"] = data.type
+        hdr["IMGTYPE"] = "spectrum"  # Temporary
         hdr["SOURCE"] = data.source
-        hdr["TEMP_SET"] = data.temperature_setpoint.to_value(u.deg_C)
-        hdr["CCDTEMP"] = data.temperature.to_value(u.deg_C)
+        hdr["TEMP_SET"] = (
+            data.temperature_setpoint.to_value(u.deg_C),
+            "Temperature set point [C].",
```

Comment:
Units come at the start of the comment string, not the end.

Note that `C` is a coulomb and not a temperature unit. There is no FITS standard way to specify degrees celsius since they expect you to use kelvin. Maybe use `degC` here if you must use celsius.

Reply:
changed to degC and moved to the start of the comment.